apps/enc.c

```diff
@@ -438,15 +455,34 @@ int enc_main(int argc, char **argv)
                     BIO_printf(bio_err, "bad magic number\n");
                     goto end;
                 }
-
                 sptr = salt;
             }
 
-            if (!EVP_BytesToKey(cipher, dgst, sptr,
-                                (unsigned char *)str,
-                                str_len, 1, key, iv)) {
-                BIO_printf(bio_err, "EVP_BytesToKey failed\n");
-                goto end;
+            if (pbkdf2 == 1) {
+                /* generate key and default iv
+                concatenated into a temporary buffer */
```

Comment:
Comment style. Should either be on a single line:

````
/* generate key and default iv concatenated into a temporary buffer */
````

Or, multi-line like this:

````
/*
 * generate key and default iv
 * concatenated into a temporary buffer
 */
````